File: Java/src/Problem250.java
import java.util.Scanner;

/**
 * Created by Ray on 2017/2/7.
 * https://www.nowcoder.com/pat/2/problem/250
 */
public class Problem250 {
    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        int a;
        while (scanner.hasNextInt()){
            a = scanner.nextInt();
            System.out.println(f(a)%3==0 ? "YES":"NO");
        }

    }

    public static int f(int a){
        if(a == 0){
            return 7;
        }
        else if(a == 1){
            return 11;
        }
        else{
            int prepre = 7;
            int pre = 11;
            for(int i = 2;i <= a;++i){
                int current = pre + prepre;
                prepre = pre;
                pre = current;
            }
            return pre;
        }
    }
}

File: PAT/PAT/Problem250.cpp
//
//  Problem250.cpp
//  PAT
//
//  Created by Ray on 2017/2/7.
//  Copyright © 2017年 Ray. All rights reserved.
//https://www.nowcoder.com/pat/2/problem/250

#include <stdio.h>
int main(){
    int arr[1000001];
    arr[0] = 7;
    arr[1] = 11;
    for(int i = 2;i < 1000001;++i){
        arr[i] = arr[i-1] + arr[i-2];
    }
    int a;
    
    while (scanf("%d",&a) != EOF) {
        if(a % 4 == 2){
            printf("Yes\n");
        }
        else{
            printf("No\n");
        }
    }
    
    return 0;
}
